bindings/azure/openai/openai.go

```diff
@@ -0,0 +1,283 @@
+/*
+Copyright 2021 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package openai
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/cognitiveservices/azopenai"
+	"github.com/dapr/components-contrib/bindings"
+	"github.com/dapr/components-contrib/metadata"
+	"github.com/dapr/kit/config"
+	"github.com/dapr/kit/logger"
+)
+
+// List of operations.
+const (
+	CompletionOperation     bindings.OperationKind = "completion"
+	ChatCompletionOperation bindings.OperationKind = "chat-completion"
```

Comment:
We normally use camelCase in Dapr

```suggestion
	ChatCompletionOperation bindings.OperationKind = "chatCompletion"
```